app/controllers/transport/departureTransportMeans/TransportMeansIdentificationController.scala

```diff
@@ -78,18 +86,17 @@ class TransportMeansIdentificationController @Inject() (
                   Future.successful(
                     BadRequest(view(formWithErrors, departureId, identificationTypeList, mode))
                   ),
-                value => redirect(mode, value, departureId)
+                value => redirect(value, departureId)
               )
         }
     }
 
   private def redirect(
-    mode: Mode,
     value: TransportMeansIdentification,
     departureId: String
   )(implicit request: MandatoryDataRequest[_]): Future[Result] =
     for {
       updatedAnswers <- Future.fromTry(request.userAnswers.set(TransportMeansIdentificationPage, value))
       _              <- sessionRepository.set(updatedAnswers)
-    } yield Redirect(navigator.nextPage(TransportMeansIdentificationPage, updatedAnswers, departureId, mode))
+    } yield Redirect(controllers.routes.CheckYourAnswersController.onPageLoad(departureId))
```

Comment:
Would prefer to keep navigation logic within the navigators if possible

Reply:
I had considered that , but my thinking was that the 3 pages on this ticket only get landed on from CYA change links and always go back to CYA, so was there a necessity to create a new Navigator just for 3 pages to go back to CYA.
However, I have now added  this for consistency